packages/replay-next/components/sources/SyntaxHighlightedLine.tsx

```diff
@@ -1,27 +1,42 @@
 import { Suspense } from "react";
 
-import { parse, parsedTokensToHtml } from "replay-next/src/suspense/SyntaxParsingCache";
+import {
+  ParsedToken,
+  parse,
+  parsedTokensToHtml,
+} from "replay-next/src/suspense/SyntaxParsingCache";
 
 import Loader from "../Loader";
 import styles from "./SyntaxHighlightedLine.module.css";
 
-type Props = { code: string; className?: string; fileExtension?: string };
+type Props = { code: string; className?: string; fileExtension?: string; tokens?: ParsedToken[] };
 
-export default function SyntaxHighlightedLine({ className = "", code, fileExtension }: Props) {
+export default function SyntaxHighlightedLine({
+  className = "",
+  code,
+  fileExtension,
+  tokens,
+}: Props) {
   return (
     <Suspense fallback={<Loader />}>
       <SyntaxHighlightedLineSuspends
         className={className}
         code={code}
         fileExtension={fileExtension}
+        tokens={tokens}
       />
     </Suspense>
   );
 }
 
-function SyntaxHighlightedLineSuspends({ code, className = "", fileExtension = ".js" }: Props) {
-  const tokens = parse(code, fileExtension);
-  const html = parsedTokensToHtml(tokens?.[0] ?? []);
+function SyntaxHighlightedLineSuspends({
+  code,
+  className = "",
+  fileExtension = ".js",
+  tokens,
+}: Props) {
+  tokens = tokens || (parse(code, fileExtension)?.[0] ?? []);
```

Comment:
Kind of a nit but this is a pretty dense statement. I personally prefer to write stuff like this out a bit more explicitly, but admittedly it's a bit subjective:
```ts
if (tokens == null) {
  const parsed = parse(code, fileExtension);
  if (parsed?.length > 0) {
    tokens = parsed[0];
  }
}
```